src/bindgen/cargo/cargo.rs

```diff
@@ -135,6 +135,28 @@ impl Cargo {
             .map(|dep| {
                 let (dep_name, dep_version) = parse_dep_string(dep);
 
+                // If a version was not specified find the only package with the name of the dependency
+                let dep_version = dep_version.or_else(|| {
+                    let mut package_iter = self.metadata.packages
+                        .iter()
+                        .filter(|p| p.name_and_version.name == dep_name && p.name_and_version.version.is_some());
+
+                    // Get the first element from the iterator as the result.
+                    let result = package_iter
+                        .next()
+                        .and_then(|p| p.name_and_version.version.as_ref().map(|f| f.as_str()));
```

Comment:
You can do `.map(|p| p.name_and_version.version.as_ref().unwrap().as_str())` if you want, as you've checked it's some...

But this may be cleaner with something like:

```rust
let versions = self.metadata.packages.iter().filter_map(|package| {
    if package.name_and_version.name != dep_name {
        return None;
    }
    p.name_and_version.version.as_ref()
});

let version = versions.next();
if versions.next().is_none() {
    version
} else {
    warn!("...");
    None
}
```

Wdyt?

Reply:
Thanks, thats much cleaner. I applied it.